test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -73,3 +178,298 @@ int suiteTearDown( int numFailures )
+    TEST_ASSERT_FALSE( returnFlags.addAcknowledgment );
+}
+
+
+/**
+ * @brief Test MQTTAgentCommand_Publish() failure case with QoS1.
+ */
+void test_MQTTAgentCommand_Publish_QoS1_failure( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTPublishInfo_t publishInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    /* Initializing QOS. */
+    publishInfo.qos = MQTTQoS1;
+
+    MQTT_GetPacketId_IgnoreAndReturn( 1 );
+    MQTT_Publish_ExpectAndReturn( &mqttAgentContext, &publishInfo, 1, MQTTSendFailed );
+
+    mqttStatus = MQTTAgentCommand_Publish( &mqttAgentContext, &publishInfo, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSendFailed, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 1, returnFlags.packetId );
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_FALSE( returnFlags.addAcknowledgment );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Subscribe() works as intended.
+ */
+void test_MQTTAgentCommand_Subscribe( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentSubscribeArgs_t subscribeArgs = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_GetPacketId_IgnoreAndReturn( 1 );
+    MQTT_Subscribe_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Subscribe( &mqttAgentContext, &subscribeArgs, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 1, returnFlags.packetId );
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_TRUE( returnFlags.addAcknowledgment );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Unsubscribe() works as intended.
+ */
+void test_MQTTAgentCommand_Unsubscribe( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentSubscribeArgs_t subscribeArgs = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_GetPacketId_IgnoreAndReturn( 1 );
+    MQTT_Unsubscribe_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Unsubscribe( &mqttAgentContext, &subscribeArgs, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 1, returnFlags.packetId );
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_TRUE( returnFlags.addAcknowledgment );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Disconnect() works as intended.
+ */
+void test_MQTTAgentCommand_Disconnect( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Disconnect_ExpectAndReturn( &mqttAgentContext, MQTTSuccess );
+    mqttStatus = MQTTAgentCommand_Disconnect( &mqttAgentContext, pUnusedArg, &returnFlags );
```

Comment:
If it's unused then I don't really see a reason to declare it
```suggestion
    mqttStatus = MQTTAgentCommand_Disconnect( &mqttAgentContext, NULL, &returnFlags );
```